x-pack/test/functional/apps/gis/es_geohashgrid_source.js

```diff
@@ -120,6 +120,30 @@ export default function ({ getPageObjects, getService }) {
         });
       });
 
+      it('should correctly re-request data', async () => {
+
+        //zoom way outside range
+        await PageObjects.gis.setView(64, 179, 5);
+        let mapboxStyle = await PageObjects.gis.getMapboxStyle();
+        expect(mapboxStyle.sources[LAYER_ID].data.features.length).to.equal(0);
+
+        //zoom back into range
+        await PageObjects.gis.setView(0, 0, 0);
+        mapboxStyle = await PageObjects.gis.getMapboxStyle();
+        expect(mapboxStyle.sources[LAYER_ID].data.features.length).to.equal(EXPECTED_NUMBER_FEATURES);
```

Comment:
Why are you comparing feature numbers instead of request times? Seems like this would be more clear if you tested before and after timestamps to see if you a new request was triggered.

Reply:
The test case should go like this...

Set map to center and zoom at the begging of a precision level. get beforeRequestTimestamp

Set map zoom to zoom into 1 or 2 levels but that does not cross a precision level. Assert a new request was not fired because beforeRequestTimestamp === currentRequestTimestamp

Set the map zoom level to a larger value that crosses the precision level. Assert a new request was fired because beforeRequestTimestamp !== currentRequestTimestamp